File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        ans = []
        for i in range(0, len(s), k):
            if(i + k - 1 < len(s)):
                ans.append(s[i:i+k])
            else:
                temp = s[i:i+k]
                temp += (k - len(temp)) * fill
                ans.append(temp)
        return ans
        

File: 2260-divide-a-string-into-groups-of-size-k/divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> ans;
        for(int i=0;i<(int)s.size();i+=k)
        {
            if(i+k-1<(int)s.size()) ans.push_back(s.substr(i,k));
            else
            {
                string temp=s.substr(i,k);
                while(temp.size()<k) temp.push_back(fill);
                ans.push_back(temp);
            }
        }
        return ans;
    }
};